languages/python/sqlalchemy-oso/sqlalchemy_oso/roles2.py

```diff
@@ -163,6 +162,113 @@ def user_in_role_query(
     return query
 
 
+def list_filter_query(kind, resource, relationships, id_field):
```

Comment:
I don't know the code enough to know if this would be helpful, so just a suggestion:

Is there a way to write the single item queries to just extend this? Would that make sense & make the code a bit easier to maintain? It would just involve adding the PK filters to the "select data" part of the sql generation.

Reply:
yeah that would probably work, will try it as a followup